bpf/lib/egress_policies.h

```diff
@@ -31,6 +31,14 @@ struct egress_gw_policy_entry *lookup_ip4_egress_gw_policy(__be32 saddr, __be32
 #endif /* ENABLE_EGRESS_GATEWAY */
 
 #ifdef ENABLE_SRV6
+struct srv6_srh {
+	struct ipv6_rt_hdr rthdr;
+	__u8 first_segment;
+	__u8 flags;
+	__u16 reserved;
+	struct in6_addr segments[0];
```

Comment:
Array of [0] isn't throwing a compiler/warning error?

Reply:
No. And that's fairly standard (in Linux, DPDK, and OVS at least) when you want to refer to space that is always after the object but that space isn't allocated as part of the object's allocation.

Reply:
Aha, I thought the syntax would be 

```
struct srv6_srh {
	struct ipv6_rt_hdr rthdr;
	__u8 first_segment;
	__u8 flags;
	__u16 reserved;
	struct in6_addr segments[];
};
```

But, TIL. Thank